pkg/kine/server/delete.go

```diff
@@ -8,32 +8,32 @@ import (
 	"go.opentelemetry.io/otel/attribute"
 )
 
-func isDelete(txn *etcdserverpb.TxnRequest) (int64, string, bool) {
+func isDelete(txn *etcdserverpb.TxnRequest) (int64, []byte, bool) {
 	if len(txn.Compare) == 1 &&
 		txn.Compare[0].Target == etcdserverpb.Compare_MOD &&
 		txn.Compare[0].Result == etcdserverpb.Compare_EQUAL &&
 		len(txn.Failure) == 1 &&
 		txn.Failure[0].GetRequestRange() != nil &&
 		len(txn.Success) == 1 &&
 		txn.Success[0].GetRequestDeleteRange() != nil {
-		return txn.Compare[0].GetModRevision(), string(txn.Success[0].GetRequestDeleteRange().Key), true
+		return txn.Compare[0].GetModRevision(), txn.Success[0].GetRequestDeleteRange().Key, true
 	}
-	return 0, "", false
+	return 0, []byte{}, false
 }
 
-func (l *LimitedServer) delete(ctx context.Context, key string, revision int64) (_ *etcdserverpb.TxnResponse, err error) {
+func (l *LimitedServer) delete(ctx context.Context, key []byte, revision int64) (_ *etcdserverpb.TxnResponse, err error) {
 	deleteCnt.Add(ctx, 1)
 	ctx, span := otelTracer.Start(ctx, fmt.Sprintf("%s.delete", otelName))
 	defer func() {
 		span.RecordError(err)
 		span.End()
 	}()
 	span.SetAttributes(
-		attribute.String("key", key),
+		attribute.String("key", string(key)),
 		attribute.Int64("revision", revision),
 	)
 
-	rev, deleted, err := l.backend.Delete(ctx, key, revision)
+	rev, deleted, err := l.backend.Delete(ctx, string(key), revision)
```

Comment:
I think we should pass the key without turning that into a string.

Reply:
yeah, we'll have to update the rest of the driver interface to use ``key []byte``, wasn't sure whether to do that here or as a separate PR.

Reply:
I've updated all the methods to use ``[]byte`` keys.